website/docs/language/functions/templatestring.mdx

```diff
@@ -4,39 +4,31 @@ description: |-
   The templatestring function takes a string from elsewhere in the module and renders its content as a template using a supplied set of template variables.
 ---
 
-# `templatestring` Function
+# `templatestring` function reference
 
--> **Note:** The `templatestring` function is intended for advanced use cases. Most use cases require only a [string template expression](/terraform/language/expressions/strings#string-templates). To render a template from a file, use the [`templatefile` function](/terraform/language/functions/templatefile).
+This topic provides reference information about the `templatestring` function. The `templatestring` function renders a string defined elsewhere in the module as a template using a set of variables. 
 
-`templatestring` renders a template given as a string value, using a supplied set of template variables.
+## Introduction
 
-```hcl
-templatestring(ref, vars)
-```
+The primary use case for the `templatestring` function is to render templates fetched as a single string from remote locations using data resources. The function enables advanced use cases where a [string template expression](/terraform/language/expressions/strings#string-templates) is insufficient, such as when the template is available from a named object declared in the current module. To render a template from a file, use the [`templatefile` function](/terraform/language/functions/templatefile).
```

Comment:
This reads to me that you can _only_ use a data source to get the template string. At least technically speaking, it looks like there's no limitation on local values, resource attributes, or even other variables. Are we advising that users _only_ use data sources with the `templatestring` function?

Reply:
I don't think that is the intent. This is an edit of the original version. I'll try a different wording.